EventFiltering/PWGLF/strangenessFilter.cxx

```diff
@@ -1044,19 +795,43 @@ struct strangenessFilter {
         hEvtvshMinPt->Fill(i + 0.5);
     }
 
-    // 2Xi trigger definition
-    if (xicounter > 1) {
-      keepEvent[2] = true;
+    // Double/triple/quad Xi trigger definition
+    if (v0sFromXiID.size() > 0) {
+      std::set<std::array<int, 2>> uniqueXis = {v0sFromXiID.begin(), v0sFromXiID.end()};
+      if (uniqueXis.size() > 1) {
+        keepEvent[2] = true;
+      }
+      if (uniqueXis.size() > 2) {
+        keepEvent[3] = true;
+      }
+      if (uniqueXis.size() > 3) {
+        keepEvent[4] = true;
+      }
     }
 
-    // 3Xi trigger definition
-    if (xicounter > 2) {
-      keepEvent[3] = true;
+    // Double Omega trigger definition
+    if (v0sFromOmegaID.size() > 0) {
+      std::set<std::array<int, 2>> uniqueOmegas = {v0sFromOmegaID.begin(), v0sFromOmegaID.end()};
+      if (uniqueOmegas.size() > 1) {
+        keepEvent[10] = true;
+      }
     }
 
-    // 4Xi trigger definition
-    if (xicounter > 3) {
-      keepEvent[4] = true;
+    // Omega + Xi trigger definition
+    if (v0sFromOmegaID.size() > 0 && v0sFromXiID.size() > 0) {
+      std::set<std::array<int, 2>> uniqueOmegas = {v0sFromOmegaID.begin(), v0sFromOmegaID.end()};
+      std::set<std::array<int, 2>> uniqueXis = {v0sFromXiID.begin(), v0sFromXiID.end()};
+      if (uniqueOmegas.size() != 0 && uniqueXis.size() != 0) {
```

Comment:
I am not sure this makes sense: if the uniqueOmegas and uniqueXis are each 1 then you will get the trigger true. I think here the condition should be:
```cpp
if (uniqueOmegas.size() > 1 || uniqueXis.size() > 1) {
  keepEvent[11] = true;
}
```